core/store/src/trie/update.rs

```diff
@@ -155,8 +169,8 @@ impl TrieUpdate {
         TrieUpdateIterator::new(self, prefix, start, Some(end))
     }
 
-    pub fn get_root(&self) -> CryptoHash {
-        self.root
+    pub fn get_root(&self) -> &CryptoHash {
```

Comment:
```suggestion
    pub fn get_root(&self) -> &StateRoot {
```
Could we do this, as we do relevant changes anyway?